10_construct-database/11_construct_county_all.qmd

```diff
@@ -0,0 +1,546 @@
+  here("04_health/data/final_data/neonatal_health_2022.csv"),
+  here("04_health/exposure-to-trauma/final/rate_injury_deaths_metric_all_county.csv"),
+  
+  # 05 local-governance
+  here("05_local-governance/descriptive-representation/data/final/descriptive_rep_denominator_county_2021.csv"),
+  here("data", "temp", "voter-turnout.csv"),
+
+  # 06 neighborhoods
+  here("data", "temp", "race-ethnicity-exposure_county.csv"),
+  here("data", "temp", "poverty-exposure_county.csv"),
+  here("06_neighborhoods/social-capital/final/social_associations_all_county.csv"),
+  here("06_neighborhoods/social-capital/final/economic_connectedness_county_2022.csv"),
+  here("06_neighborhoods/environment/data/final/environment_county_longitudinal.csv"),
+  
+  # 07 safety
+  here("07_safety/final/rates_crime_county_all.csv"),
+  here("07_safety/final/rate_arrests_county_all.csv"),
+  
+  # 08 education
+  here("08_education/data/final/metrics_preschool_county_all_longitudinal.csv"),
+  here("08_education/SEDA_all_county.csv"),
+  here("08_education/data/final_data/meps_county_2020.csv"),
+  here("08_education/data/final/metrics_college_county_all_longitudinal.csv"),
+  here("08_education/final/digital_access_county_all.csv"),
+  
+  # 09 employment
+  here("09_employment/data/final/metrics_employment_county_all_longitudinal.csv"),
+  here("09_employment/metrics_wage_ratio_2022.csv")
+)
+
+# variables erroneously included in the data sets
+extra_variables <- c(
+  "subgroup", 
+  "subgroup_type",
+  "ratio_population_pc_physician_flag_low_pop",
+  "rate_injury_death_se", 
+  "rate_injury_death_crude"
+  
+)
+
+```
+
+This code loads each file and then combines them using `left_join()`. The first file is a population file, so there should be 3,143 counties per year. 
+
+```{r}
+db <- filepaths |>
+  map(
+    .f = ~read_csv(.x) |> 
+      select(-any_of(c("state_name", "county_name", extra_variables)))
+  ) |>
+  reduce(left_join, by = c("year", "state", "county"))
+
+
+```
+
+## Quality and Completeness
+
+### Data Quality Flags
+
+This section summarizes data quality flags for variables and years.
+
+```{r}
+db |>
+  select(ends_with("_quality")) |>
+  pivot_longer(everything(), names_to = "variable", values_to = "quality") |>
+  count(variable, quality) |>
+  filter(!is.na(quality)) |>
+  print(n = 100)
+
+db |>
+  select(year, ends_with("_quality")) |>
+  pivot_longer(-year, names_to = "variable", values_to = "quality") |>
+  count(year, quality) |>
+  print(n = 100)
+
+```
+
+### Full Database
+
+Evaluate the quality variables. 
+
+```{r}
+db |>
+  select(ends_with("_quality")) |>
+  assert(in_set(1, 2, 3), everything())
+
+```
+
+### 01 Financial Well-Being
+
+```{r}
+db |>
+  assert(
+    within_bounds(0, Inf), 
+    pctl_income_20, 
+    pctl_income_50, 
+    pctl_income_80
+  ) |>
+  assert(
+    within_bounds(0, Inf), 
+    share_debt_coll, 
+    share_debt_coll_ub, 
+    share_debt_coll_lb
+  ) |>
+  quiet()
+
```

Comment:
The pctl_income variables have CIs - why are we not using the test bounds function on these variables? Im guessing it's becuase for these vlaues there is the possibility of a CI being Inf even when the variable is included and it would throw off the test.

Reply:
This was an oversight. Good catch!